packages/nimble-components/src/table/styles.ts

```diff
@@ -25,6 +28,16 @@ export const styles = css`
 
     :host {
         height: 480px;
+        ${tableFitRowsHeight.cssCustomProperty}: calc(var(--ni-private-table-scroll-height) + ${controlHeight});
```

Comment:
I think it would be worthwhile to add a unit test that ensures the `tableFitRowsHeight` token gets updated after a call that changes the number of rows. The Chromatic tests do a good job of covering the render behavior and they indirectly cover `setData()`. But explicit unit tests would be valuable so that it's tested closer to the implementation. You could also think about whether we'd want separate tests for each the cases that could impact this (e.g. calling `setData()` vs `setRecordHierarchyOptions()`).

Reply:
Added some tests that made sure the table's height is correct when styled with that token when data is set, grouping is applied, and when rows are collapsed.